elixir/apps/api/lib/api/controllers/identity_controller.ex

```diff
@@ -0,0 +1,71 @@
+defmodule API.IdentityController do
+  use API, :controller
+  alias API.Pagination
+  alias Domain.Auth
+
+  action_fallback API.FallbackController
+
+  # List Identities
+  def index(conn, %{"actor_id" => actor_id} = params) do
+    subject = conn.assigns.subject
+    list_opts = Pagination.params_to_list_opts(params)
+
+    with {:ok, actor} <- Domain.Actors.fetch_actor_by_id(actor_id, subject),
+         {:ok, identities, metadata} <- Auth.list_identities_for(actor, subject, list_opts) do
+      render(conn, :index, identities: identities, metadata: metadata)
+    end
+  end
+
+  # Create an Identity
+  def create(conn, %{
+        "actor_id" => actor_id,
+        "provider_id" => provider_id,
+        "identity" => params
+      }) do
+    subject = conn.assigns.subject
+
+    params =
+      Map.put_new(
+        params,
+        "provider_identifier_confirmation",
+        Map.get(params, "provider_identifier")
+      )
+
+    with {:ok, actor} <- Domain.Actors.fetch_actor_by_id(actor_id, subject),
+         {:ok, provider} <- Auth.fetch_provider_by_id(provider_id, subject),
+         true = valid_provider?(provider),
```

Comment:
This will crash the code and API consumer won't have a proper error message:
```
iex(1)> with true = false do
...(1)>   :ok
...(1)> end
** (MatchError) no match of right hand side value: false
    (stdlib 6.0) erl_eval.erl:652: :erl_eval.expr/6
    iex:1: (file)
```

Reply:
Ah, yeah, that makes sense now that you point it out.  How would you handle this?  Would you just have an explicit check inside the `with` block that verifies that the provider is a `manual` provisioner or is there a slicker/more elegant way of handling this?

Reply:
You can either check for `:manual` value right there or just do `true <- valid_provider?(provider)` and add an `else` clause that will convert `false` to something meaningful for the client.